.github/workflows/publish-npm-packages.yml

```diff
@@ -0,0 +1,125 @@
+        name: Sync the WordPress release branch
+        runs-on: ubuntu-latest
+        outputs:
+            plugin_version: ${{ steps.get_version.outputs.plugin_version }}
+            plugin_release_branch: ${{ steps.get_version.outputs.plugin_release_branch }}
+
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+              with:
+                  token: ${{ secrets.GUTENBERG_TOKEN }}
+
+            - name: Compute plugin release branch
+              id: get_version
+              run: |
+                  PLUGIN_VERSION=$(jq --raw-output '.version' package.json)
+                  echo "::set-output name=plugin_version::$(echo $PLUGIN_VERSION)"
+                  IFS='.' read -r -a PLUGIN_VERSION_ARRAY <<< "$PLUGIN_VERSION"
+                  PLUGIN_RELEASE_BRANCH="release/${PLUGIN_VERSION_ARRAY[0]}.${PLUGIN_VERSION_ARRAY[1]}"
+                  echo "::set-output name=plugin_release_branch::$(echo $PLUGIN_RELEASE_BRANCH)"
+
+            - name: Configure git user name and email
+              run: |
+                  git config user.name "Gutenberg Repository Automation"
+                  git config user.email gutenberg@wordpress.org
+
+            - name: Switch to WordPress release branch
+              run: |
+                  git fetch --depth=1 origin "$WORDPRESS_RELEASE_BRANCH"
+                  git checkout "$WORDPRESS_RELEASE_BRANCH"
+
+            - name: Sync the WordPress release branch with the changes from the latest plugin release
+              if: |
+                  github.event.inputs.release_type == 'latest' ||
+                  github.event.inputs.release_type == 'next'
+              run: |
+                  git rm -r .
+                  git checkout origin/"${{ steps.get_version.outputs.plugin_release_branch }}" -- .
+
+            - name: Commit the changes to the WordPress release branch
+              if: |
+                  github.event.inputs.release_type == 'latest' ||
+                  github.event.inputs.release_type == 'next'
+              run: |
+                  git commit -m "Merge changes published in the Gutenberg plugin ${{ steps.get_version.outputs.plugin_release_branch }} branch"
+                  git push --set-upstream origin "$WORDPRESS_RELEASE_BRANCH"
+
+    update-packages:
+        name: Update Packages
+        runs-on: ubuntu-latest
+        needs: sync-wordpress-branch
+
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
+              with:
+                  ref: ${{ env.WORDPRESS_RELEASE_BRANCH }}
+
+            - name: Use desired version of NodeJS
+              uses: actions/setup-node@38d90ce44d5275ad62cc48384b3d8a58c500bb5f # v2.2.2
+              with:
+                  node-version: 14
+                  cache: npm
+
+            - name: Update package versions
+              run: |
```

Comment:
We already handle version updates in the existing script:

https://github.com/WordPress/gutenberg/blob/c9ba45839c0e2521d507d95757ab1bfdb0a307ec/bin/plugin/commands/common.js#L85

https://github.com/WordPress/gutenberg/blob/c9ba45839c0e2521d507d95757ab1bfdb0a307ec/bin/plugin/commands/packages.js#L115

Reply:
Yep! 👍 Following the example of the other release related GHA workflows, I've started translating them to Shell script. This should allow us to avoid installing a ton of npm dependencies and speed up the process (and eventually to retire the JS counterpart) 🙂

Reply:
I'm not sure it's worth the extra work. To publish WordPress packages to npm you need to run:
- `npm install` (to have at least a subset of dependencies to be able to build packages).
- `npm run build:packages` (uses tools that output - TypeScript types, CSS, JS).
- Lerna commands like `npx lerna version` and `npx lerna publish`.

Reply:
Right, makes sense. I kinda forgot we need to build the packages anyway in order to publish them.

Reply:
OTOH, the JS script is currently interactive, isn't it? (Requires confirmation a couple of times.) So there'd need to be at least a bit of work done to allow non-interactive usage if we wanted to use it here 🤔

Reply:
> the JS script is currently interactive, isn't it? (Requires confirmation a couple of times.) So there'd need to be at least a bit of work done to allow non-interactive usage if we wanted to use it here 

Yes, we need to skip all prompts with some CLI arg like `--ci`. It's definitely also possible with Lerna commands used internally.